src/Qowaiv.CodeAnalysis.CSharp/Rules/DefinePropertiesAsNotNullable.cs

```diff
@@ -20,22 +21,40 @@ private void ReportProperty(SyntaxNodeAnalysisContext context)
 
     private void ReportRecord(SyntaxNodeAnalysisContext context)
     {
-        var types = context.Node.Cast<RecordDeclarationSyntax>()
-            .ParameterList?.Parameters.Select(p => p.Type)
-            .OfType<TypeSyntax>();
+        if (context.Node.Cast<RecordDeclarationSyntax>().ParameterList is { } pars)
+        {
+            foreach (var type in pars.Parameters.Select(p => p.Type))
```

Comment:
No IEnumerable<T>.ForEach(Action<T> a) extension?

Reply:
That only exists on list.